bot/extensions/fun/weather.py

```diff
@@ -0,0 +1,104 @@
+from geopy.geocoders import Nominatim
+from timezonefinder import TimezoneFinder
+from pytz import country_timezones, timezone
+from datetime import datetime
+from discord.ext.commands import Cog, command
+from requests import get
+from discord import Embed
+from string import capwords
+
+
+class WeatherCog(Cog, name="Weather", description="get current weather information from a city"):
+    def __init__(self, bot):
+        self.bot = bot
+    
+    @command(name='weather', help='Show weather information in your city', usage="{city}")
+    async def weather(self, ctx, *city_input):
+        city = capwords(" ".join(city_input))
+
+        # initialize Nominatim API
+        geolocator = Nominatim(user_agent="geoapiExercises")
+        
+        # getting Latitude and Longitude
+        location = geolocator.geocode(city)
+
+        # pass the Latitude and Longitude
+        # into a timezone_at
+        # and it return timezone
+        timezone_finder = TimezoneFinder()
+
+        result = timezone_finder.timezone_at(
+            lng=location.longitude, 
+            lat=location.latitude)
+        timezone_city = datetime.now(timezone(result))
+        timezone_city = timezone_city.strftime('%m/%d/%Y %H:%M')
+
+        api_key = "441df3a5cadc2498e093c0367cae6817"
+        base_url = "http://api.openweathermap.org/data/2.5/weather?"
+
+        # complete_url to retreive weather info
+        complete_url = base_url + "appid=" + api_key + "&q=" + city
+        response = get(complete_url)
+        data_weather = response.json()
+
+        # Now data_weather contains list of nested dictionaries, 
+        # check the value of "cod" if key is equal to 200
+        # it means the city is found otherwise, city is not found
+        if data_weather["cod"] == 200:
+            icon_id = data_weather["weather"][0]["icon"]    # get image icon id
```

Comment:
This goes for all bellow. Having a comment that says `get X` isn't a useful comment because it's easy to understand what happens here. You want to add a comment when something needs more clarification because you are unable to do it directly with the code you write.